OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -118,8 +132,33 @@ public override Activity Tick(Actor self)
 					return NextActivity;
 				}
 
-				// Fly towards the last known position
-				QueueChild(self, new Fly(self, target, WDist.Zero, lastVisibleMaximumRange, checkTarget.CenterPosition, Color.Red), true);
+				// Fly towards the desired facing (if CanHover) or last known position
+				if (aircraft.Info.CanHover)
+					aircraft.SetPosition(self, aircraft.CenterPosition + aircraft.FlyStep(desiredFacing));
```

Comment:
Maybe this should queue `HeliFly` as a `ChildActivity` instead of doing all the positioning stuff itself?

Reply:
The problem with that, at least in this particular activity, is that it needs to check every tick whether it's close enough to the target now, so using a ChildActivity for movement would complicate things.

I agree that this activity shouldn't call SetPosition and TickFacing directly, though, and should at least use `Fly.FlyToward` from #16483 instead.

Reply:
But `HeliFly` already takes care of that, doesn't it? Why can't we just call `move.MoveWithinRange`, just like `AttackFollow` does?

Reply:
Thanks for pointing me in the right direction, adressed in #16483.